### PR TITLE
enable podpresets with service catalog

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/bootstrap/docker/openshift/helper.go [hunk 1]
```diff
@@ -733,6 +733,15 @@ func (h *Helper) updateConfig(configDir string, opt *StartOptions) error {
...
 
 	if opt.ServiceCatalog {
+
+		if cfg.AdmissionConfig.PluginConfig == nil {
+			cfg.AdmissionConfig.PluginConfig = map[string]configapi.AdmissionPluginConfig{}
+		}
+
+		cfg.AdmissionConfig.PluginConfig["PodPreset"] = configapi.AdmissionPluginConfig{
+			Configuration: &configapi.DefaultAdmissionConfig{Disable: false},
+		}
```
Comment (bparees) [inline pkg/bootstrap/docker/openshift/helper.go]: @pmorie you see anything wrong with this?  it did not appear to make podpresets work for me.

**Comment by bparees** [comment]
@spadgett @jwforres this will PR enable podpresets once i get it working and merge it.

**Comment by bparees** [comment]
ok this works with one small issue..  it seems like secret keys with a dash do not get injected properly because dashes aren't valid in env variable names but they are valid as keys in secrets.

@jim-minter at a minimum this means we should not be using dashes in our templates when we expose values (we currently have things like template.openshift.io/expose-admin-password": "{.data['database-admin-password').  Whether we should go so far as to error on them during binding is a question worth asking.

**Comment by bparees** [comment]
@pmorie ptal

**Comment by jim-minter** [comment]
@bparees It seems like it's the SC's decision to take these values and put them into the envfrom section of the podpreset.  It could in principle create a configmap object and mount it in (in which case hyphens would be fine).  Also note https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/apis/servicecatalog/types.go#L383 .

So I suppose it would probably be expedient for us to s/-/_/g everything after `expose-` in our examples for now, but I'm not convinced we should error on it.

**Comment by jim-minter** [comment]
Also see https://github.com/openshift/origin/pull/14824

**Comment by bparees** [comment]
> So I suppose it would probably be expedient for us to s/-/_/g everything after expose- in our examples for now, but I'm not convinced we should error on it.

fair enough.

**Comment by pmorie** [comment]
LGTM

**Comment by bparees** [comment]
[merge][severity:lowrisk]

**Comment by bparees** [comment]
[merge][severity:lowrisk]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/14129
flake https://github.com/openshift/origin/issues/13108
flake https://github.com/openshift/origin/issues/14043

**Pull request merged**